npp.py

```diff
@@ -12,26 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-## ------- Set up our list of numbers -------
-S = [25, 7, 13, 31, 42, 17, 21, 10]
-
-## ------- Set up our QUBO dictionary -------
+## ------- import packages -------
+from dwave.system import DWaveSampler, EmbeddingComposite
 
 # TODO:  Add code here to define your QUBO dictionary
+def define_QUBO(S):
 
-## ------- Run our QUBO on the QPU -------
-# TODO:  Choose QPU parameters
-chainstrength = 1
-numruns = 10
+    Q = {}
 
-# Run the QUBO on the solver from your config file
-from dwave.system import DWaveSampler, EmbeddingComposite
+    # Add QUBO construction here
+    
+    return Q
+
+# TODO:  Choose QPU parameters in the following function
+def run_on_QPU(Q, sampler):
+
+    chainstrength = 1 # update
+    numruns = 1 # update
+
+    sample_set = sampler.sample_qubo(Q, chain_strength=chainstrength, num_reads=numruns)
+
+    return sample_set
+
+
+## ------- Main program -------
+if __name__ == "__main__":
+
+    ## ------- Set up our list of numbers -------
+    S = [25, 7, 13, 31, 42, 17, 21, 10]
+
+    # TODO: Enter your token here
+    token = 'Your-Token-Here'
+
+    ## ------- Set up our QUBO dictionary -------
+
+    Q = define_QUBO(S)
```

Comment:
I still think this line should be `Q = get_qubo(S)`.

However, I would understand if what I am suggesting does not make sense in a classroom context